aptos-move/framework/aptos-framework/sources/vesting_without_staking.move

```diff
@@ -0,0 +1,1174 @@
+            left = left - amount;
+        });
+
+        // Send any remaining "dust" (leftover due to rounding error) to the withdrawal address.
+        if (coin::value(&coins) > 0) {
+            aptos_account::deposit_coins(vesting_contract.withdrawal_address, coins);
+        } else {
+            coin::destroy_zero(coins);
+        };
+
+        emit_event(
+            &mut vesting_contract.distribute_events,
+            DistributeEvent {
+                admin: vesting_contract.admin,
+                vesting_contract_address: contract_address,
+            },
+        );
+    }
+
+    /// Remove the lockup period for the vesting contract. This can only be called by the admin of the vesting contract.
+    /// Example usage: If admin find shareholder suspicious, admin can remove it.
+    public entry fun remove_shareholder(admin: &signer, contract_address: address, shareholder_address: address) acquires VestingContract {
+        let vesting_contract = borrow_global_mut<VestingContract>(contract_address);
+        verify_admin(admin, vesting_contract);
+        let shareholders = &mut vesting_contract.shareholders;
+        simple_map::remove(shareholders, &shareholder_address);
+        assert!(simple_map::contains_key(shareholders, &shareholder_address), 0);
+    }
+
+    /// Terminate the vesting contract and send all funds back to the withdrawal address.
+    public entry fun terminate_vesting_contract(admin: &signer, contract_address: address) acquires VestingContract {
+        assert_active_vesting_contract(contract_address);
+
+        vest(contract_address);
+
+        let vesting_contract = borrow_global_mut<VestingContract>(contract_address);
+        verify_admin(admin, vesting_contract);
+
+        vesting_contract.state = VESTING_POOL_TERMINATED;
+
+        emit_event(
+            &mut vesting_contract.terminate_events,
+            TerminateEvent {
+                admin: vesting_contract.admin,
+                vesting_contract_address: contract_address,
+            },
+        );
+    }
+
+    /// Withdraw all funds to the preset vesting contract's withdrawal address. This can only be called if the contract
+    /// has already been terminated.
+    public entry fun admin_withdraw(admin: &signer, contract_address: address) acquires VestingContract {
+        let vesting_contract = borrow_global<VestingContract>(contract_address);
+        assert!(vesting_contract.state == VESTING_POOL_TERMINATED, error::invalid_state(EVESTING_CONTRACT_STILL_ACTIVE));
+
+        let vesting_contract = borrow_global_mut<VestingContract>(contract_address);
+        verify_admin(admin, vesting_contract);
+        let withdrawn_coins = coin::balance<AptosCoin>(contract_address);
+        let coins = coin::withdraw<AptosCoin>(admin, withdrawn_coins);
+        let amount = coin::value(&coins);
+        if (amount == 0) {
+            coin::destroy_zero(coins);
+            return
+        };
```

Comment:
You need to retrieve the signer from `vesting_contract` as `vesting_signer` and just do `coin:;transfer` of entire balance. Then call `coin::destroy_zero`. You are trying to `withdraw` from `admin` but who is `admin`? Who are supposed to transfer from the vesting contract.

Reply:
I have updated this. I think we do not need to call coin::destroy_zero as do coin transfer do not create the coin.